_posts/2021/2021-08-02-dont-call-replacement-classes-new.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Don't call a new version of something 'new' when refactoring"
+description: "A opinionated naming subtlety"
+layout: article
+category: ruby
+image:
+  base: '2021/dont-call-replacement-classes-new'
+  alt: "New"
+  credit: Nick Fewings
+  source: "https://unsplash.com/photos/1SsUquHPNT8"
+
+---
+
+In long-lived applications, we often want to rewrite or refactor an approach that we have previously taken.
```

Comment:
*an approach _to do a particular thing_ (not super keen on that and maybe it's not needed..)

Reply:
```suggestion
In long-lived applications, we often want to rewrite or refactor an approach that we have previously taken to solving a problem.
```